### PR TITLE
OpInfo porting for torch.maximum / torch.minimum / torch.fmax / torch.fmin

### PR DESCRIPTION
**Pull request by Kiyosora**
Related #54261

This PR ports the method_tests() entries of following operators to OpInfo.
- torch.maximum
- torch.minimum
- torch.fmax
- torch.fmin

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -954,6 +954,7 @@ def sample_inputs_max_min_binary(op_info, device, dtype, requires_grad):
         ((S, S, S), (S,),),
         ((S,), (S, S, S),),
         ((S, 1, S), (S, S),),
+        ((S, S), (S, S),),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:957]: This is a clever reuse of the existing function.

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -3099,6 +3100,30 @@ def reference_lgamma(x):
            dtypesIfCUDA=all_types_and(torch.float16, torch.bfloat16, torch.bool),
            supports_out=False,
            sample_inputs_func=sample_inputs_max_min_reduction_no_dim,),
+    OpInfo('maximum',
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3103]: Good choice to do all 4 of these at the same time.

**Review by mruberry — APPROVED** [review]
Cool! Thanks @Kiyosora!

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -3144,6 +3145,30 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
+           op=torch.maximum,
+           dtypes=all_types_and(torch.float16, torch.bfloat16, torch.bool),
+           dtypesIfCPU=all_types_and(torch.float16, torch.bfloat16, torch.bool),
+           dtypesIfCUDA=all_types_and(torch.float16, torch.bfloat16, torch.bool),
```
Comment (imaginary-person) [inline torch/testing/_internal/common_methods_invocations.py:3152]: In an OpInfo, `dtypesIfCPU` and `dtypesIfCUDA` only need to be assigned, if they're different from the `dtypes` list. I see that these redundancies arose from looking at the `OpInfos` of `min` & `max`, that are close to these ones. I had introduced those redundancies after looking at the OpInfos of `masked_scattered`, etc, but would be removing them.

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -3144,6 +3145,30 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
+           sample_inputs_func=sample_inputs_max_min_binary,),
+    OpInfo('minimum',
+           op=torch.minimum,
+           dtypes=all_types_and(torch.float16, torch.bfloat16, torch.bool),
+           dtypesIfCPU=all_types_and(torch.float16, torch.bfloat16, torch.bool),
+           dtypesIfCUDA=all_types_and(torch.float16, torch.bfloat16, torch.bool),
+           sample_inputs_func=sample_inputs_max_min_binary,),
+    OpInfo('fmax',
```
Comment (imaginary-person) [inline torch/testing/_internal/common_methods_invocations.py:3160]: I think alphabetical order is usually followed while placing `OpInfo`s.

File: torch/testing/_internal/common_methods_invocations.py (hunk 3 above)
Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:3152]: That would be a nice cleanup, you're right, but let's get this merged.

File: torch/testing/_internal/common_methods_invocations.py (hunk 4 above)
Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:3160]: Alphabetical order is ideal. We can fix it later ;)

**Comment by Kiyosora** [comment]
Hi @mruberry, @imaginary-person . Thank you so much for reviewing this PR! I have fixed the point you suggested before, please take a look if it's convenient for you.

**Comment by imaginary-person** [comment]
@Kiyosora: I wasn't reviewing per se, but GitHub considers anyone who comments (on someone else's PR) a reviewer. :)

**Comment by mruberry** [comment]
Importing to land now.

**Comment by mruberry** [comment]
Merged. Let's just let the tests run to make sure I did it properly.

**Pull request closed without merging**